pkg/runner/step.go

```diff
@@ -94,6 +96,20 @@ func runStepExecutor(step step, stage stepStage, executor common.Executor) commo
 		}
 		logger.Infof("\u2B50 Run %s %s", stage, stepString)
 
+		// Prepare and clean Runner File Commands
+		actPath := ActPath
+		outputFileCommand := path.Join("workflow", "outputcmd.txt")
+		stateFileCommand := path.Join("workflow", "statecmd.txt")
+		(*step.getEnv())["GITHUB_OUTPUT"] = path.Join(actPath, outputFileCommand)
+		(*step.getEnv())["GITHUB_STATE"] = path.Join(actPath, stateFileCommand)
```

Comment:
Could we set these once inside `run_context.go` instead of setting them to the same value whenever each step runs.  Then update this code to just clear out the files using the paths grabbed from the step env?

Reply:
Okay, so I fiddled around with this on a real action, and it looks like GITHUB_OUTPUT is set to a different path for each step (like `/home/github/github-actions/_work/_temp/_runner_file_commands/set_output_18bd9c4c-7e30-40eb-b659-4a0a3be207b4`)

![image](https://user-images.githubusercontent.com/1055021/201966177-d7d19169-22c4-4ad1-8c7b-fc5660d69389.png)

I also tried overriding the value using an `env:` block but it appears to have been ignored.  So maybe leaving this logic here and having it reset/clear out on every step is the right way to go. Or if you want to be even MORE like the real thing, then set it to a random file name on each run.

Reply:
And it looks like they actually use the same guid (maybe a generated step ID from each run) for all of the file commands and then merge stuff in later.

```
  echo "Writing to $GITHUB_OUTPUT"
  echo "Writing to $GITHUB_STATE"
  echo "Writing to $GITHUB_ENV"

Writing to /home/github/github-actions/_work/_temp/_runner_file_commands/set_output_67a16ed1-fc66-[4](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:4)fc9-9a4a-a601a4abdf[5](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:5)0
Writing to /home/github/github-actions/_work/_temp/_runner_file_commands/save_state_[6](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:6)[7](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:7)a16ed1-fc66-4fc[9](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:9)-9a4a-a601a4abdf50
Writing to /home/github/github-actions/_work/_temp/_runner_file_commands/set_env_67a[16](https://github.com/Unity-Technologies/mp-groups/actions/runs/3472080444/jobs/5802425679#step:4:17)ed1-fc66-4fc9-9a4a-a601a4abdf50
```